src/main/resources/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildTrigger/config.jelly

```diff
@@ -52,5 +52,11 @@
     <f:entry title="Phrase to request a (re-)build" field="ciBuildPhrases">
       <f:textbox default="test this please"/>
     </f:entry>
+    <f:entry title="Approve PR on build success" field="approveOnBuildSuccessful">
+      <f:checkbox default="false"/>
+    </f:entry>
+    <f:entry title="Mark PR with Needs Work on build failure" field="needsWorkOnBuildFailure">
+      <f:checkbox default="false"/>
+    </f:entry>
```

Comment:
Please move those entries above `Merge PR if build is successful`. That way, the users would get the idea that the marking would happen before the attempted merge. It's impossible to mark a PR as approved after the merge.

Actually, it's an interesting question whether the merge would be attempted if the PR is not mergeable without the approval from Jenkins. I'm afraid it won't be. It can be implemented separately.

Reply:
Good question. On one hand we can block such attempt from plugin side or we can based on bitbucket configuration whether it will allow this action or not.

Reply:
done